airflow/models/dag.py

```diff
@@ -2436,6 +2446,21 @@ def set_edge_info(self, upstream_task_id: str, downstream_task_id: str, info: Ed
         """
         self.edge_info.setdefault(upstream_task_id, {})[downstream_task_id] = info
 
+    def validate_schedule_and_params(self):
+        """
+        Validates & raise exception if there are any Params in the DAG which neither have a default value nor
+        have the null in schema['type'] list, but the DAG have a schedule_interval which is not None.
+        """
+        if not self.timetable.can_run:
+            return
+
+        for k, v in self.params.items():
+            # As type can be an array, we would check if `null` is a allowed type or not
+            if v.default is None and ('type' not in v.schema or "null" not in v.schema['type']):
+                raise AirflowException(
+                    'DAG Schedule must be None, if there are any required params without default values'
+                )
```

Comment:
```suggestion
            if v.default is None and ("type" not in v.schema or "null" not in v.schema["type"]):
                raise AirflowException(
                    "DAG Schedule must be None, if there are any required params without default values"
                )
```

nit -- for consistency in just these 3 lines